### ISSUE TITLE
[Feature request] Enhance q-layout-drawer with additional state information

### ISSUE THREAD

**Issue by dsl101** [quasarframework/quasar#2410]
I'm struggling to get a clean design inside a q-layout-drawer which has dynamic content (a menu with changing contents, etc.). What I really need to know, in order to control the sizing and positioning of the content is exactly how big the drawer is when opened, and whether it opened in desktop or overlay mode. This is important because its height is different when in overlay mode (it covers the q-toolbar-title area, for example), so aligning something to the _bottom_ of the drawer is error-prone without knowing this.

For my specific use case, I would need to know the state of `mobileView` when it changes—perhaps a simple additional event could be created on changes to that data element? Or perhaps there's a more elegant way to provide access to the current state of the drawer by making the model an object not a boolean. I realise this is a breaking change though, unless the component can check on initialisation the type of bound variable that was passed in?
```
    <q-layout-drawer v-model="simple" />
    <q-layout-drawer v-model="complex" />
.
.
.
    data () {
      return {
        simple: false,
        complex: {
          showing: false,
          mobileView: false
        }
      }
    }
````

**Comment by dsl101** [comment]
Actually, it turns out if you open the drawer on a narrow desktop browser window (so that it opens in overlay mode) then drag the window wider, it stays open and the data element which I need is `belowBreakpoint`, since `mobileView` isn't updated until the next time the drawer is manually closed and opened again. So maybe I need an event on `belowBreakpoint` changing...

**Comment by rstoenescu** [comment]
Hi,

Are you looking for styling content only? Because the QLayoutDrawer gets the `q-layout-drawer-mobile` when it's in "Mobile" mode.

**Comment by dsl101** [comment]
Well, I want to change the styling of _another component_ when the drawer is open—e.g. making a flex grid in the main page go from `col-4` to `col-6` when the drawer is open in 'desktop' mode, but not change when opened in 'mobile' mode. The reason being that the usable width of the flex area changes when the drawer opens in desktop mode, but not in mobile mode.

In the end I had to inject the q-layout, and compute the column classes based on using the same width logic that q-layout-drawer uses to determine which mode to use. But of course that could easily get out of sync with q-layout-drawer if that logic changes.

**Comment by rstoenescu** [comment]
Event `on-layout` is emitted when drawer takes up space on layout or releases it:

```
<q-layout-drawer @on-layout="onLayout" ....
onLayout (space) {
   // space is Boolean (true - drawer takes up space, false - drawer does not take up space)
```

Available in future v0.17.9
